src/mlpack/methods/ann/quantization_utils.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef MLPACK_METHODS_ANN_QUANTIZATION_QUANTIZATION_UTILS_HPP
+#define MLPACK_METHODS_ANN_QUANTIZATION_QUANTIZATION_UTILS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * Find the quantization scale factor for converting weights to a target type.
+ *
+ * @tparam TargetType The target type for quantization (e.g., int8_t).
+ * @tparam MatType The source matrix type.
+ * @param weights The weights to be quantized.
+ * @return The scale factor for quantization.
+ */
+template<typename TargetType, typename MatType>
+double FindQuantizationScale(const MatType& weights)
+{
+  double maxAbs = arma::abs(weights).max();
+  double targetTypeMax = std::numeric_limits<TargetType>::max();
+  return targetTypeMax / maxAbs;
+}
+
+/**
+ * Quantize weights from one type to another using the given scale factor.
+ *
+ * @tparam SourceMatType The source matrix type.
+ * @tparam TargetMatType The target matrix type for quantization.
+ * @param sourceWeights The original weights.
+ * @param targetWeights The quantized weights (output).
+ * @param scaleFactor The scale factor for quantization.
+ */
+template<typename SourceMatType, typename TargetMatType>
+void QuantizeWeights(const SourceMatType& sourceWeights,
+                     TargetMatType& targetWeights,
+                     double scaleFactor)
+{
+  targetWeights = arma::conv_to<TargetMatType>::from(
+      arma::clamp(sourceWeights * scaleFactor, 
+                  std::numeric_limits<typename TargetMatType::elem_type>::min(),
+                  std::numeric_limits<typename TargetMatType::elem_type>::max()));
```

Comment:
So this is just one strategy for quantization, but ideally in the longer-term we would want to support more.  One way to do this is to turn the functions in this file into a class with a couple members, like `SimpleQuantization` or whatever you want to call it.  Then, `Quantize()` can take a template parameter indicating the strategy to use (by default `SimpleQuantization`), and inside of `Quantize()` all you need to do is call, e.g., `quantizationStrategy.QuantizeWeights()` or similar.